gulpfile.js

```diff
@@ -300,17 +299,16 @@ gulp.task('test-create-rollup', (cb) => {
 
 // This task updates the codegen_mappings.json file in sync with the azure-rest-api-specs public repo.
 gulp.task('sync-mappings-with-repo', (cb) => {
-  if (!specRepoDir) {
-    return cb(new Error('Please provide --spec-repo-dir <Absolute path to the directory where the azure-rest-api-specs is cloned.>'));
+  if (!specRoot) {
+    return cb(new Error('Please provide --spec-root <Absolute path to the local clone of the .../azure-rest-api-specs/specification repository folder.>'));
   }
-  let specDir = `${specRepoDir}/specification`;
-  const dirs = fs.readdirSync(specDir).filter(f => fs.statSync(`${specDir}/${f}`).isDirectory());
+  const dirs = fs.readdirSync(specRoot).filter(f => fs.statSync(`${specRoot}/${f}`).isDirectory());
   let newlyAdded = [];
   let originalProjectCount = Object.keys(mappings).length;
   for (let rp of dirs) {
-    if (rp.toLowerCase() === 'intune' || rp.toLowerCase() === 'azsadmin' || rp.toLowerCase() === 'timeseriesinsights') continue;
-    let rm = `${specRepoDir}/specification/${rp}/resource-manager`;
-    let dp = `${specRepoDir}/specification/${rp}/data-plane`;
+    if (rp.toLowerCase() === 'common-types' || rp.toLowerCase() === 'intune' || rp.toLowerCase() === 'azsadmin' || rp.toLowerCase() === 'timeseriesinsights') continue;
```

Comment:
Your goal here I'm sure isn't to rewrite this but I'd sure appreciate something like:

```js
const servicesToSkip = ['common-types', 'intune', 'azsadmin', ...];
// using indexOf() === -1 to mean "not contains"
if (servicesToSkip.indexOf(rp.toLowerCase()) === -1) {
  // actually do stuff
}
```
instead of the long line with a `continue` tucked in the end.

Reply:
I've extracted this to an array variable, as you suggested.